sarif/src/test/java/com/jetbrains/qodana/sarif/StreamParseCorrectnessTest.java

```diff
@@ -87,4 +100,27 @@ public void testStreamParseReadsSanityCorrectly() throws IOException {
             assertFalse(actualResults.hasNext());
         }
     }
+
+    private static void assertRunPropertyBag(Run expected, StreamingRun actual, Integer runIndex) throws IOException {
+        if (expected.getProperties() != null) {
+            actual.getReader(reader -> {
+                Iterator<Object> expectedSanityResults = ((List<Object>) expected.getProperties().getOrDefault(
+                        "qodana.sanity.results",
+                        Collections.emptyList()
+                )).iterator();
+                Iterator<Result> actualSanityResults = SarifUtil.lazyReadResultsFromLocation(
+                        reader,
+                        new ResultLocation.InProperties(runIndex, "qodana.sanity.results")
+                );
+                while (expectedSanityResults.hasNext()) {
+                    assertTrue(actualSanityResults.hasNext());
+                    Result expectedSanityResult = gson.fromJson(gson.toJson(expectedSanityResults.next()), Result.class);
```

Comment:
nit, don't need to parse strings again (ofc irrelevant here, but good to know for other places 😅)
```suggestion
                    Result expectedSanityResult = gson.fromJson(gson.toJsonTree(expectedSanityResults.next()), Result.class);
```